### PR TITLE
Update TensorFlow to 1.14.0

### PR DESCRIPTION
**Pull request by gitanshu**
This commit updates a lot of deprecated imports to tf.compat.v1 imports, and some other method updates like (tf.log -> tf.math.log) and more. There are still more deprecations, but merging it as-is is still useful since it also brings python 3.7 support.

### CODE REVIEW

File: docs/user/implement_algo_advanced.rst [hunk 1]
```diff
@@ -92,7 +92,7 @@ extra diagnostic information as well as supporting recurrent policies):
             'action',
             batch_dims=1,
         )
-        advantage_var = tf.placeholder('advantage')
+        advantage_var = tf.compat.v1.placeholder('advantage')
```
Comment (ryanjulian) [inline docs/user/implement_algo_advanced.rst]: does it throw deprecation warnings if you don't use the compat.v1 modules?

Comment (gitanshu) [reply docs/user/implement_algo_advanced.rst]: Yes

Comment (ryanjulian) [reply docs/user/implement_algo_advanced.rst]: alright

**Comment by ryanjulian** [comment]
you might need to wait for #774 for this one, because it is exceeding travis max runtime (because it has to rebuild the docker).

**Comment by ryanjulian** [comment]
do the tests pass locally?

**Comment by gitanshu** [comment]
> do the tests pass locally?

Yup

**Comment by zequnyu** [comment]
It seems I had the same issue with #777.

**Comment by ryanjulian** [comment]
#774 should fix it

File: .travis.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 language: python
 
-python: "3.5"
+python: "3.7"
```
Comment (ryanjulian) [inline .travis.yml]: this doesn't affect the tested python version, because evaluations happen in docker not in travis

**Comment by zequnyu** [comment]
Do we support tensorflow 1.13? It seems pip doesn't have `tensorflow<1.13,>=1.12.0` now.

**Comment by ryanjulian** [comment]
@zequnyu can you post the error message you got which suggests PyPI doesn't have TF1.13 anymore?

The objective of this PR is to upgrade to TF1.14. We only support one version of TF at a time (the API moves too quickly to do anything else).

**Comment by gitanshu** [comment]
@ryanjulian this happens on python 3.7 only, since 3.7 isn't added as supported version on TF Pypi before version 1.14
Merging this PR should fix that.

**Comment by ryanjulian** [comment]
@gitanshu  sorry -- what happens only in Py37? -- i didn't follow Zequn's original question

**Comment by gitanshu** [comment]
Basically if you do `pip install tensorflow<1.13.1` on python 3.7, pypi says `No matching distribution found`. 1.13.1 was the first version to officially support python 3.7 through pypi

**Comment by ryanjulian** [comment]
@gitanshu can you push this over the finish line today? I think #773 will make it a gnarly refactor if we don't.

**Comment by ryanjulian** [comment]
@gitanshu @zequnyu ah. great -- so it sounds like this PR will allow us to support Py3.5-3.7?

**Comment by gitanshu** [comment]
> @gitanshu @zequnyu ah. great -- so it sounds like this PR will allow us to support Py3.5-3.7?

py36-py37 -> yes. For py35, I am seeing a problem in local where the resolved version of atari-py (0.1.15) is unavailable for py35, but somehow seems to work fine on CI with the same version of python 3.5.6

**Comment by gitanshu** [comment]
This PR is ready to go though, as soon as CI starts behaving again

**Pull request merged**